webapp/src/Controller/API/ContestController.php

```diff
@@ -330,62 +330,113 @@ public function setBannerAction(Request $request, string $cid, ValidatorInterfac
-     *                 description="Force overwriting the start_time even when in next 30s",
+     *                 description="Force overwriting the start_time even when in next 30s or the scoreboard_thaw_time when already set or too much in the past",
      *                 type="boolean",
+     *             ),
+     *             @OA\Property(
+     *                 property="scoreboard_thaw_time",
+     *                 description="The new unfreeze (thaw) time of the contest",
+     *                 type="string",
+     *                 format="date-time"
      *             )
      *         )
      *     )
      * )
      * @OA\Response(
      *     response="200",
-     *     description="Contest start time changed successfully",
+     *     description="Contest object if it changed",
      *     @OA\JsonContent(
-     *         type="string"
+     *         allOf={
+     *             @OA\Schema(ref=@Model(type=Contest::class)),
+     *             @OA\Schema(ref="#/components/schemas/Banner")
+     *         }
      *     )
      * )
+     * @OA\Response(
+     *     response="204",
+     *     description="The change was successful"
+     * )
      */
-    public function changeStartTimeAction(Request $request, string $cid): Response
+    public function changeTimesAction(Request $request, string $cid): Response
     {
         $contest  = $this->getContestWithId($request, $cid);
-        $now      = Utils::now();
+        $now      = (int)Utils::now();
         $changed  = false;
         if (!$request->request->has('id')) {
-            $response = new JsonResponse('Missing "id" in request.', Response::HTTP_BAD_REQUEST);
-        } elseif (!$request->request->has('start_time')) {
-            $response = new JsonResponse('Missing "start_time" in request.', Response::HTTP_BAD_REQUEST);
-        } elseif ($request->request->get('id') != $contest->getApiId($this->eventLogService)) {
-            $response = new JsonResponse('Invalid "id" in request.', Response::HTTP_BAD_REQUEST);
-        } elseif (!$request->request->has('force') &&
-            $contest->getStarttime() != null &&
-            $contest->getStarttime() < $now + 30) {
-            $response = new JsonResponse('Current contest already started or about to start.',
-                                         Response::HTTP_FORBIDDEN);
-        } elseif ($request->request->get('start_time') === null) {
-            $this->em->persist($contest);
-            $contest->setStarttimeEnabled(false);
-            $response = new JsonResponse('Contest paused :-/.', Response::HTTP_OK);
-            $this->em->flush();
-            $changed = true;
-        } else {
-            $date = date_create($request->request->get('start_time'));
-            if ($date === false) {
-                $response = new JsonResponse('Invalid "start_time" in request.', Response::HTTP_BAD_REQUEST);
+            return new JsonResponse('Missing "id" in request.', Response::HTTP_BAD_REQUEST);
```

Comment:
There is probably a difference, but I am fine with raising an exception, seems we do that in other places.

Reply:
Apparantly it helps dev mode but also it will make the code coverage slightly more trustworthy. It doesn't punish for tests not covering such exceptions.